### PR TITLE
🚑️(api) fix sync error using MySQL

### LINKED ISSUE #1956: MultipleObjectsReturned error during sync course from LMS to Richie
## Bug Report

**Problematic Behavior**
Error synchronizing course from LMS to Richie after adding a course code.

**Expected behavior/code**
Richie API raises:
```
get() returned more than one Course -- it returned 2!
ERROR 2023-04-06 17:06:39,973 log 42 140255949047616 Internal Server Error: /api/v1.0/course-runs-sync/
Traceback (most recent call last):
  File "/usr/local/lib/python3.10/site-packages/django/core/handlers/exception.py", line 47, in inner
    response = get_response(request)
  File "/usr/local/lib/python3.10/site-packages/django/core/handlers/base.py", line 181, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/usr/local/lib/python3.10/site-packages/django/views/decorators/csrf.py", line 54, in wrapped_view
    return view_func(*args, **kwargs)
  File "/usr/local/lib/python3.10/site-packages/django/views/generic/base.py", line 70, in view
    return self.dispatch(request, *args, **kwargs)
  File "/usr/local/lib/python3.10/site-packages/rest_framework/views.py", line 509, in dispatch
    response = self.handle_exception(exc)
  File "/usr/local/lib/python3.10/site-packages/rest_framework/views.py", line 469, in handle_exception
    self.raise_uncaught_exception(exc)
  File "/usr/local/lib/python3.10/site-packages/rest_framework/views.py", line 480, in raise_uncaught_exception
    raise exc
  File "/usr/local/lib/python3.10/site-packages/rest_framework/views.py", line 506, in dispatch
    response = handler(request, *args, **kwargs)
  File "/usr/local/lib/python3.10/site-packages/rest_framework/decorators.py", line 50, in handler
    return func(*args, **kwargs)
  File "/usr/local/lib/python3.10/site-packages/richie/apps/courses/api.py", line 270, in sync_course_runs_from_request
    sync_course_run(request.data)
  File "/usr/local/lib/python3.10/site-packages/richie/apps/courses/api.py", line 163, in sync_course_run
    course = Course.objects.get(
  File "/usr/local/lib/python3.10/site-packages/django/db/models/manager.py", line 85, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
  File "/usr/local/lib/python3.10/site-packages/django/db/models/query.py", line 439, in get
    raise self.model.MultipleObjectsReturned(
richie.apps.courses.models.course.Course.MultipleObjectsReturned: get() returned more than one Course -- it returned 2!
``` 

**Steps to Reproduce**
1. Create a new course
2. Add a code to the course
3. And then the bug happens!

**Environment**
- Richie version: `2.21.0`
- Platform: Linux

**Additional context/Screenshots**
When I run this on python shell - `python manage.py shell`:
```python
>>> from richie.apps.courses.models import Course
>>> course_code='FF_PP'
>>> Course.objects.get(code=course_code, extended_object__publisher_is_draft=True, extended_object__node__parent__cms_pages__course__isnull=True, )
Traceback (most recent call last):
  File "<console>", line 1, in <module>
  File "/usr/local/lib/python3.10/site-packages/django/db/models/manager.py", line 85, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
  File "/usr/local/lib/python3.10/site-packages/django/db/models/query.py", line 439, in get
    raise self.model.MultipleObjectsReturned(
richie.apps.courses.models.course.Course.MultipleObjectsReturned: get() returned more than one Course -- it returned 2!
``` 

URL: https://www.nau.edu.pt/pt/curso/primeiros-passos-em-folha-de-calculo-formulas-e-funcoes-nivel-1/

Nevertheless, I found out that the synchronization is running and the number of course enrollments have been updated automatically.

### PR DESCRIPTION
**Pull request by igobranco**
Fix error `MultipleObjectsReturned` when synchronize course runs, when using MySQL as Richie database.
fixes #1956

### CODE REVIEW

File: src/richie/apps/courses/api.py [hunk 1]
```diff
@@ -159,18 +159,19 @@ def sync_course_run(data):
...
-        course = Course.objects.get(
+    course = (
+        Course.objects.filter(
             code=course_code,
             extended_object__publisher_is_draft=True,
             # Exclude snapshots
             extended_object__node__parent__cms_pages__course__isnull=True,
         )
-    except Course.DoesNotExist as exc:
+        # distinct fixes MySQL returning same course twice
```
Comment (sampaccoud) [inline src/richie/apps/courses/api.py:169]: I don't think it is related to mysql, but to the join in the clause:
`extended_object__node__parent__cms_pages`
```suggestion
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -8,6 +8,10 @@ Versioning](https://semver.org/spec/v2.0.0.html).
 
 ## [Unrealeased]
 
+### Fixed
+
+- Fix error `MultipleObjectsReturned` during sync using MySQL
```
Comment (sampaccoud) [inline CHANGELOG.md:13]:
```suggestion
- Fix error `MultipleObjectsReturned` during synchronization due to missing `distinct`
```

**Review by sampaccoud — APPROVED** [review]

File: src/richie/apps/courses/api.py [hunk 3]
```diff
@@ -159,18 +159,19 @@ def sync_course_run(data):
...
+    )
+    if not course:
         # Create the course page in draft
-        raise ValidationError(
-            {"resource_link": [f"Unknown course: {course_code:s}."]}
-        ) from exc
+        raise ValidationError({"resource_link": [f"Unknown course: {course_code:s}."]})
 
     # Instantiate a new draft course run
     draft_course_run = CourseRun(
```
Comment (sampaccoud) [inline src/richie/apps/courses/api.py:177]: I think it can be reproduced in a test

Comment (jbpenrath) [reply src/richie/apps/courses/api.py:177]: @igobranco I push a commit where there is a test which reveals the issue.
In fact, the problem occurred once the page has been published... 😳 (The `extended_object__node__parent__cms_pages__course__isnull=True` targets both draft and published pages)

File: CHANGELOG.md (hunk 2 above)
Comment (igobranco) [reply CHANGELOG.md]: Done.

File: src/richie/apps/courses/api.py (hunk 1 above)
Comment (igobranco) [reply src/richie/apps/courses/api.py]: Yes, it also works! passed the tests 😀. So I'm changing to your solution.

File: src/richie/apps/courses/api.py (hunk 3 above)
Comment (igobranco) [reply src/richie/apps/courses/api.py]: Your test didn't has impact on the problem.
Nevertheless, I've still included in the code.

**Comment by igobranco** [comment]
In resume:
The query needs a `distinct`. 
It was failing when the course page has a parent "courses" page. In production sites, all course pages has a "courses" parent page, but we didn't include any test that verify this scenario. I've included a new test that calls a synchronization API for a course with a parent "courses" page. This new test prevent that the problem reappear. 

@jbpenrath  can you review again?

**Review by jbpenrath — APPROVED** [review]
LGTM ✅

**Pull request merged**